client/my-sites/hosting/staging-site-card/use-staging-sync.ts

```diff
@@ -0,0 +1,64 @@
+import { useMutation, UseMutationOptions, useIsMutating } from '@tanstack/react-query';
+import { useCallback, useMemo } from 'react';
+import wp from 'calypso/lib/wp';
+
+type MutationVariables = number;
+
+interface MutationResponse {
+	message: string;
+}
+
+interface MutationError {
+	code: string;
+	message: string;
+}
+
+export const PUSH_TO_STAGING = 'push-to-staging-site-mutation-key';
+
+const pushSyncMutation =
+	( sitePath: string, key: string ) =>
+	< X, Y, Z, W >( firstSiteId: number, options: UseMutationOptions< X, Y, Z, W > ) => {
```

Comment:
What is the X, Y, Z and W?

Reply:
This is in order to provide the parameters in UseMutationOptions, to make typescript happy. UseMutationOptions use some internal types,  that we override underneath, so this is why this is there.

Reply:
Thanks for the explanation. Could we make it consistent with other usages of `UseMutationOptions`?